routes/api/signup.router.js

```diff
@@ -33,10 +36,31 @@ signup.post("/", async (req, res) => {
 
     const hashedPassword = await bcrypt.hash(password, 10);
 
-    const newUser = new User({ email, password: hashedPassword, avatarURL }); // Dodajemy avatarURL do danych nowego użytkownika
+    const verificationToken = uuidv4();
+
+    const newUser = new User({
+      email,
+      password: hashedPassword,
+      avatarURL,
+      verificationToken,
+    });
 
     await newUser.save();
 
+    const verificationLink = `${req.protocol}://${req.get(
+      "host"
+    )}/users/verify/${verificationToken}`;
```

Comment:
Możesz zrobić mały helper function który bedzie tworzył taki link, bo skoro go uzywasz w dwoch miejscach to wydaje się być dobrym kandydatem.